### PR TITLE
FIX Restore adding sent emails to test session state

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripeltd/product-issues/issues/609

Requires https://github.com/silverstripe/silverstripe-framework/pull/10558 be merged first

Fix for https://github.com/silverstripe/silverstripe-admin/actions/runs/3293582405/jobs/5430233136#step:12:483
```
    And there should be an email to "admin@example.org" titled "Your password reset link"             # SilverStripe\BehatExtension\Context\EmailContext::thereIsAnEmailFromToTitled()
      Could not find email to "admin@example.org" titled "Your password reset link".
```

### CODE REVIEW

**Comment by michalkleiner** [comment]
@emteknetnz the framework PR has been merged, not sure if anything else is needed here. The tests here were already green.

**Comment by emteknetnz** [comment]
Look like I made an incorrect assumption that the framework PR needed to be merged first

File: src/Utility/TestMailer.php [hunk 1]
```diff
@@ -27,6 +31,24 @@ public function __construct(
         $this->testSessionEnvironment = TestSessionEnvironment::singleton();
     }
 
+    public function send(RawMessage $message, Envelope $envelope = null): void
+    {
+        parent::send($message, $envelope);
+        if (!is_a($message, Email::class)) {
+            throw new InvalidArgumentException('$message must be a ' . Email::class);
+        }
```
Comment (GuySartorelli) [inline src/Utility/TestMailer.php]:
```suggestion
```
This is already checked and thrown in `parent::send()`

File: src/Utility/TestMailer.php [hunk 2]
```diff
@@ -27,6 +31,24 @@ public function __construct(
...
+        /** @var Email $email */
+        $email = $message;
+        $data = $this->createData($email);
```
Comment (GuySartorelli) [inline src/Utility/TestMailer.php:39]:
```suggestion
        $data = end($this->emailsSent);
```
`parent::send()` already created this data for us.

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

File: src/Utility/TestMailer.php (hunk 1 above)
Comment (emteknetnz) [reply src/Utility/TestMailer.php]: Updated

File: src/Utility/TestMailer.php (hunk 2 above)
Comment (emteknetnz) [reply src/Utility/TestMailer.php:39]: Can't do this, $this->emailsSent is private, not protected

It's a very low cost operation and has no side-effects, so I'm not fussed about the duplicate method call

Comment (GuySartorelli) [reply src/Utility/TestMailer.php:39]: Fair enough

**Review by GuySartorelli — APPROVED** [review]

**Pull request merged**